README.md

```diff
@@ -3,13 +3,53 @@
+  Algo $B_2$: $50(1,000,000)+100$ = $50000100$ or .00005% of Algo $B_2$<br><br>
+  You can see from the percentages that as the input size grows the lower order terms and constants have a much smaller impact on the amount of operations the algorithm does. So in practice you need to know how impactful your lower order terms and constants will be based on individual circumstances. Yes, for large datasets the asymptotic complexity is what's going to dictate which algorithm is better. But if the dataset is under that threshold like I've shown, that can be misleading between practice and theory
+
+This also goes into seeing a algorithm with a smaller time complexity but a large constant going slower then an algorithm with a worse time complexity and smaller constant. 
+
+ Hardware as an issue. Asymptotic analysis does not take into account for hardware specifics like your cache and or memory hierarchy. Running an algorithm and getting a certain Big-O bound on one machine could be significantly different on a different machine.
+    
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+Having $O(log_2n)$ as the asymptotic complexity for searching for a specific element in a binary search tree of 1000 elements in 5 seconds has the same logarithmic growth for 10,000 elements. So, $ 5 secs / $log_2(1000)$ = $x / log_2(10,000)$ = $5 * log_2(10,000) / log_2(1000)$. 5*13.29= 66.44  $log_2(1000)$ = 9.97. (66.44/9.97) = 6.7 secs to find the same element in the tree with 10,000 items.
+
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
+  
+  1. There could be an issue with memory so the computer is running out of memory. Having fragmented data because the memory is being swapped or paged to a disk drive if there isn't enough RAM for the larger data set. As the input grows the amount of memory needed is also increased. Yes, the growth is not linear but logaritmic. Even if the growth is not 10 times more the growth added logarithmic can still be the straw that breaks the camels back. Causing more overhead and issues with transfering data between RAM and disk space.
+  2. Maybe the impact of an unbalanced tree isn't as noticable with 1000 elements as it is with 10,000 elements. If it was unbalanced and it was making linear comparisons, you would expect the 10,000 elements to take 50 seconds. But the unbalance might not be uniform and have larger strucural inefficencies due to the increase in list size. This could account for the increase in time being 100 secs and not the linear 50 secs.
```

Comment:
Still not following you here. Linear is the worst case, i.e. there can't be anything that is worse. And that only gets you to 50 seconds.